tiledb/sm/fragment/fragment_identifier.h

```diff
@@ -0,0 +1,133 @@
+/**
+ * @file tiledb/sm/fragment/fragment_identifier.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2024 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file defines class FragmentID.
+ */
+
+#ifndef TILEDB_FRAGMENT_ID_H
+#define TILEDB_FRAGMENT_ID_H
+
+#include "tiledb/common/status.h"
+#include "tiledb/sm/filesystem/uri.h"
+
+namespace tiledb::sm {
+
+using timestamp_range_type = std::pair<uint64_t, uint64_t>;
+
+/**
+ * The possible fragment name versions.
+ *
+ * ONE: __uuid_t1{_t2}
+ * TWO: __t1_t2_uuid
+ * THREE: __t1_t2_uuid_version
+ *
+ * @ref `format_spec/timestamped_name.md`
+ */
+enum class FragmentNameVersion { ONE, TWO, THREE };
+
+/**
+ * Validate, parse and handle the different components of a fragment identifer.
+ */
+class FragmentID {
```

Comment:
Should we add a comparison operator?

Reply:
No. We're not expanding the scope of this PR.